hikey.mk

```diff
@@ -66,33 +66,23 @@ mcuimage-cleaner:
 ################################################################################
 # ARM Trusted Firmware
 ################################################################################
-arm-tf: mcuimage optee-os edk2
+ARM_TF_FLAGS ?= \
 	CFLAGS="-O0 -gdwarf-2" \
-	CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)" \
 	BL32=$(OPTEE_OS_BIN) \
 	BL33=$(EDK2_BIN) \
 	NEED_BL30=yes \
 	BL30=$(MCUIMAGE_BIN) \
-	make -C $(ARM_TF_PATH) \
-	       -j`getconf _NPROCESSORS_ONLN` \
-	       DEBUG=$(DEBUG) \
-	       PLAT=hikey \
-	       SPD=opteed \
-	       all fip
+	DEBUG=$(DEBUG) \
+	PLAT=hikey \
+	SPD=opteed
+
+arm-tf: mcuimage optee-os edk2
+	CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)" \
+	$(MAKE) -C $(ARM_TF_PATH) $(ARM_TF_FLAGS) all fip
 
 arm-tf-clean:
-	CFLAGS="-O0 -gdwarf-2" \
-        CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)" \
-        BL32=$(OPTEE_OS_BIN) \
-        BL33=$(EDK2_BIN) \
-        NEED_BL30=yes \
-        BL30=$(MCUIMAGE_BIN) \
-        make -C $(ARM_TF_PATH) \
-               -j`getconf _NPROCESSORS_ONLN` \
-               DEBUG=$(DEBUG) \
-               PLAT=hikey \
-               SPD=opteed \
-               clean
+	CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)" \
+	$(MAKE) -C $(ARM_TF_PATH) $(ARM_TF_FLAGS) clean
```

Comment:
This is not correct, because for ARM TF some flags such as `CFLAGS` have to be passed as environment variables (otherwise they override the values set in the Makefile and you lose some required flags), while some others can indeed be passed as make arguments.

With the above I get the following error:
```
  DEPS    build/hikey/release/bl31/plat_io_storage.d
plat/hikey/plat_io_storage.c:32:26: fatal error: arch_helpers.h: No such file or directory
 #include <arch_helpers.h>
```

Here is what works for me:
```
ARM_TF_FLAGS ?= \
	CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)" \
	BL32=$(OPTEE_OS_BIN) \
	BL33=$(EDK2_BIN) \
	NEED_BL30=yes \
	BL30=$(MCUIMAGE_BIN) \
	DEBUG=$(DEBUG) \
	PLAT=hikey \
	SPD=opteed

ARM_TF_EXPORTS ?= CFLAGS="-O0 -gdwarf-2"

arm-tf: mcuimage optee-os edk2
	export $(ARM_TF_EXPORTS) ; \
	$(MAKE) -C $(ARM_TF_PATH) $(ARM_TF_FLAGS) all fip

arm-tf-clean:
	export $(ARM_TF_EXPORTS) ; \
	$(MAKE) -C $(ARM_TF_PATH) $(ARM_TF_FLAGS) clean
```